homeassistant/components/nibe_heatpump/__init__.py

```diff
@@ -163,25 +220,26 @@ async def _async_update_data(self) -> dict[int, Coil]:
         async def read_coil(coil: Coil):
             return await self.connection.read_coil(coil)
 
-        callbacks: dict[int, list[CALLBACK_TYPE]] = defaultdict(list)
-        for update_callback, context in list(self._listeners.values()):
-            assert isinstance(context, set)
-            for address in context:
-                callbacks[address].append(update_callback)
-
         result: dict[int, Coil] = {}
 
-        for address, callback_list in callbacks.items():
+        for address in self.context_callbacks.keys():
```

Comment:
We can remove `.keys()`.

Reply:
Will fix on next lib update since this should change slightly.